antlr_ast.py

```diff
@@ -68,18 +86,17 @@ def _from_fields(cls, visitor, ctx, fields=None):
         return cls(ctx, **field_dict)
 
     def _get_field_names(self):
-        od = OrderedDict([(el.split('->')[-1], None) for el in self._fields])
-        return list(od)
+        return list([el.split('->')[-1] for el in self._fields])
```

Comment:
I think this change might include a field names multiple times, if there are multiple mappings to it.

For example,

```
SomeNode(AstNode):
    _fields = ['op->expression', 'exp->expression']
```

looks it might now return ['expression', 'expression']. AFAIK the reason an OrderedDict is used, is so that the field_names are returned in the order they're first encountered...

```
In [1]: from collections import OrderedDict

In [2]: OrderedDict([('a', 1), ('b', 2), ('a', 3)])
Out[2]: OrderedDict([('a', 3), ('b', 2)])
```

I think this could cause issues when checking selectors, since the visitor uses _fields to visit each field (so would visit them twice). (not totally sure though)

Reply:
You're right about what the OrderedDict does. [I now found that it was added in an earlier commit to fix a bug](https://github.com/datacamp/antlr-ast/commit/4c126f202a27e322f55b7ce1b5b8dba62745aed3), but it doesn't provide details.
The function is only used to dump a tree. [Walking over a tree skips fields if they are already present.](https://github.com/datacamp/antlr-ast/blob/5b9251c33b4aeb284efedc8c499ab719ac8a3f05/antlr_ast.py#L62-L64)
Where are multiple mappings to the same field used (where could it have caused the bug at the time?)

As the change wasn't necessary, I reverted it.